api/src/main/java/org/openmrs/api/impl/ProgramWorkflowServiceImpl.java

```diff
@@ -74,35 +74,45 @@ public Program saveProgram(Program program) throws APIException {
 			throw new APIException("Program.concept.required", (Object[]) null);
 		}
 		
-		// ProgramWorkflow
 		for (ProgramWorkflow workflow : program.getAllWorkflows()) {
-			
-			if (workflow.getConcept() == null) {
-				throw new APIException("ProgramWorkflow.concept.required", (Object[]) null);
-			}
-			if (workflow.getProgram() == null) {
-				workflow.setProgram(program);
-			} else if (!workflow.getProgram().equals(program)) {
-				throw new APIException("Program.error.contains.ProgramWorkflow", new Object[] { workflow.getProgram() });
-			}
-			
-			// ProgramWorkflowState
+			ensureProgramIsSet(workflow, program);
 			for (ProgramWorkflowState state : workflow.getStates()) {
-				
-				if (state.getConcept() == null || state.getInitial() == null || state.getTerminal() == null) {
-					throw new APIException("ProgramWorkflowState.requires", (Object[]) null);
-				}
-				if (state.getProgramWorkflow() == null) {
-					state.setProgramWorkflow(workflow);
-				} else if (!state.getProgramWorkflow().equals(workflow)) {
-					throw new APIException("ProgramWorkflow.error.contains.state", new Object[] { workflow.getProgram() });
-				}
+				ensureProgramWorkflowIsSet(state, workflow);
 			}
 		}
 		return dao.saveProgram(program);
 	}
 	
 	/**
+	 * Helper method for SaveProgram to check the workflow's program
+	 */
+	private void ensureProgramIsSet(ProgramWorkflow workflow, Program program) throws APIException {
+		if (workflow.getConcept() == null) {
```

Comment:
don't worry, your commit is well done. these are just minor things.

the method is private so will not be documented in our public API docs. there can of course be a javadoc for a private method if needed but if we write short expressive methods and give meaningful names to methods, parameters, variables there is no need for comments. the code will speak for itself. this is the idea of this refactoring. you can browse around in this code base and you will find parts that have comments explaining what the code around it does. this to me shows that the code can/should be refactored as it needs explanation. browse around and maybe you'll find improvements you would like to propose :)